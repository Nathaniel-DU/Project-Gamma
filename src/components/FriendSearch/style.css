button.blue-button {
    background-color: #00203FFF;
    /* margin: auto; */
    min-width: 90%;
    cursor: pointer;
    margin-right: 0.25em;
    margin-top: 0.7em;
    /* padding: 	0.938em; */
    border: none;
    border-radius: 10px;
    color: white;
    margin-bottom: 0.9em;
}