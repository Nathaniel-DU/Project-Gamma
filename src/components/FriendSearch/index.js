import React from 'react';
import axios from 'axios';
import './style.css';

const validEmailRegex = RegExp(/^(([^<>()[\].,;:\s@"]+(\.[^<>()[\].,;:\s@"]+)*)|(".+"))@(([^<>()[\].,;:\s@"]+\.)+[^<>()[\].,;:\s@"]{2,})$/i);

const validateForm = (errors) => {
    let valid = true;
    Object.values(errors).forEach(
      (val) => val.length > 0 && (valid = false)
    );
    return valid;
}

class FriendSearch extends React.Component {

    state = {
        email: null,
        errors: {
            email: ''
        },
        inviteRes: ''
    }

    handleChange = event => {
        event.preventDefault();
        const { name, value } = event.target;
        let errors = this.state.errors;

        switch (name) {
            case 'email': 
              errors.email = 
                validEmailRegex.test(value)
                  ? ''
                  : 'Email is not valid!';
              break;
              default:
            break;
        }
        this.setState({errors, [name]: value});
        
    }

    handleSubmit = event => {
        event.preventDefault();
        if(validateForm(this.state.errors)) {
        const friend = {
            email: this.state.email,
        }
        axios.post(`/user/friendinvite`, friend)
        .then(res => {
            this.setState({inviteRes: res.data});
        });
        console.info('Valid Search')
        }else{
        console.error('Invalid Search')
        }
        event.target.reset();
    }
  
    render () {
      return (
        <div>
            <form onSubmit={this.handleSubmit}>
                <h4>Add Friend:</h4>
                <label htmlFor="search-friend"></label>
                <input id='search-friend' name='email' type="text" placeholder="Enter email" onChange={this.handleChange}/>
                {this.state.errors.email.length > 0 && 
                <span className='error'>{this.state.errors.email}</span>}
                {this.state.inviteRes.length > 0 && 
                <span className='error'>{this.state.inviteRes}</span>}
                <button className='blue-button' type='submit'>Invite</button>
            </form>
        </div>
      );
    }
}
  
export default FriendSearch;