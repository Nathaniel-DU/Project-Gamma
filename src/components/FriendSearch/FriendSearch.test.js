import React from 'react';
import { shallow } from 'enzyme';
import FriendSearch from './index';

describe('FriendSearch', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<FriendSearch debug />);
        expect(component).toMatchSnapshot();
    })
})