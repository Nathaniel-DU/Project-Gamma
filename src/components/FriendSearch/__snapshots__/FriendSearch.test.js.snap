// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FriendSearch should render correctly in "debug" mode 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <h4>
      Add Friend:
    </h4>
    <label
      htmlFor="search-friend"
    />
    <input
      id="search-friend"
      name="email"
      onChange={[Function]}
      placeholder="Enter email"
      type="text"
    />
    <button
      className="blue-button"
      type="submit"
    >
      Invite
    </button>
  </form>
</div>
`;
